.github/workflows/ci-cd.yml

```diff
@@ -105,6 +105,31 @@ jobs:
           overwrite: true
           file_glob: true
 
+      - name: Check if GraphQL Schema Changed
+        if: matrix.os == 'linux' && matrix.arch == 'amd64'
+        id: changed-files-specific
+        uses: tj-actions/changed-files@v29
+        with:
+          files: |
+            pkg/extensions/search/schema.graphql
+
+      - name: Generate GraphQL Introspection JSON
+        if: steps.changed-files-specific.outputs.any_changed == 'true' && matrix.os == 'linux' && matrix.arch == 'amd64'
+        run: |
+          echo "Schema has changed"
+          bin/zot-linux-amd64 serve examples/config-search.json &
+          sleep 10
+          mkdir introspection
+          curl -X POST -H "Content-Type: application/json" -d @.github/introspection-query.json http://localhost:5000/v2/_zot/ext/search | jq > introspection/introspection-result.json
+          pkill zot
+
+      - name: Upload GraphQL Introspection JSON
+        uses: actions/upload-artifact@v3
+        if: steps.changed-files-specific.outputs.any_changed == 'true' && matrix.os == 'linux' && matrix.arch == 'amd64'
+        with:
+          name: introspection-query.json
+          path: introspection/introspection-result.json
```

Comment:
ok, this makes sense.

Reply:
Renamed the artifact name to introspection-result to avoid confusion